### ISSUE TITLE
.NET 7: LibraryImportAttribute

### ISSUE THREAD

**Issue by PMeira** [PMeira/dss_sharp#14]
From [P/Invoke source generation | Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/standard/native-interop/pinvoke-source-generation):

> .NET 7 introduces a [source generator](https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) for P/Invokes in C# which recognizes the [LibraryImportAttribute](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.libraryimportattribute).

.NET adoption is all over the place in terms of versions (a lot of people stuck with old Framework versions), but `LibraryImportAttribute` could reduce the overhead. This could be especially good for a future alternative API that would mirror [dss::obj](https://dss-extensions.org/dss_capi/index.html). For the moment we're already faster than the COM implementation in a lot of the API.

.NET 7 is still in the preview stage, so we have time to test before the final release (expected by the end of the year).

**Comment by PMeira** [comment]
Related: https://devblogs.microsoft.com/dotnet/performance_improvements_in_net_7/

**Comment by PMeira** [comment]
Turns out just using `LibraryImport` doesn't actually help much for DSS Sharp. The reason is that we already migrated to DSS C-API's GR interface, which API buffers to optimize some operations. 

Allowing something like `ArrayPool` would also be nice but I don't see it fitting in the current ("classic") API. 

[Source generators](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) still might be useful for our future plans, including the new API. But, since there are thousands of functions in the API, we have to weight the benefits, that is, if the assembly gets too bloated, it will actually be slower. That's another issue though, so closing this for now.

For numeric data DSS Sharp was already quite faster than using the COM DLL (or OpenDSSDirect.DLL) due to the architecture of the official APIs, so this is not really a point of concern.
